timesketch/lib/analyzers/hashr_lookup.py

```diff
@@ -0,0 +1,303 @@
+"""Sketch analyzer plugin for hashR lookup."""
+import logging
+import sys
+
+from math import ceil
+from flask import current_app
+import sqlalchemy as sqla
+from timesketch.lib.analyzers import interface, manager
+
+logger = logging.getLogger('timesketch.analyzers.hashR')
+
+
+class HashRLookup(interface.BaseAnalyzer):
+    """Analyzer for HashRLookup."""
+
+    NAME = 'hashr_lookup'
+    DISPLAY_NAME = 'hashR lookup'
+    DESCRIPTION = ('Lookup file hash values in the known hashes database '
+                   'generated by hashR.')
+
+    DEPENDENCIES = frozenset()
+
+    zerobyte_file_counter = 0
+    known_hash_event_counter = 0
+    unique_known_hash_counter = 0
+    hashr_conn = None
+    add_source_attribute = None
+    query_batch_size = None
+    DEFAULT_BATCH_SIZE = 50000
+
+    def __init__(self, index_name, sketch_id, timeline_id=None):
+        """Initialize The Sketch Analyzer.
+
+        Args:
+            index_name: Opensearch index name
+            sketch_id: Sketch ID
+            timeline_id: Timeline ID, default is None
+        """
+        self.index_name = index_name
+        super().__init__(index_name, sketch_id, timeline_id=timeline_id)
+
+    def connect_hashr(self):
+        """Connect to the hashR postgres database.
+
+        Returns:
+          True: If connection is setup and successfully tested.
+          Error Message: If the connection cannot be setup or tested.
+
+        Raises:
+          Exception:  Raises an exception if the database connection details
+                      cannot be loaded from the timesketch.conf file.
+        """
+
+        # Note: Provide connection infos in the data/timesketch.conf file!
+        db_user = current_app.config.get('HASHR_DB_USER')
+        db_pass = current_app.config.get('HASHR_DB_PW')
+        db_address = current_app.config.get('HASHR_DB_ADDR')
+        db_port = current_app.config.get('HASHR_DB_PORT')
+        db_name = current_app.config.get('HASHR_DB_NAME')
+        self.add_source_attribute = current_app.config.get(
+            'HASHR_ADD_SOURCE_ATTRIBUTE')
+        self.query_batch_size = int(current_app.config.get(
+            'HASHR_QUERY_BATCH_SIZE', self.DEFAULT_BATCH_SIZE))
+
+        if not all(config_param for config_param in [
+                db_user, db_pass, db_address, db_port,
+                db_name, self.add_source_attribute]):
+            msg = ('The hashR analyzer is not able to load the hashR database '
+                   'information from the timesketch.conf file. Please make sure'
+                   ' to uncomment the section and provide the required '
+                   'connection details!')
+            sys.tracebacklimit = 0
```

Comment:
This exception message is displayed in Timesketch when an analyser fails. Suppressing the traceback info reduces the information surfaced to the user on the web interface to the relevant message since the traceback is not relevant for this situation and would only make the error message unreadable.